Add mdbook based user guide (#619)
PyO3/maturin: guide/src/bindings.md

# Bindings

> maturin supports several kind of bindings, some of them are automatically

Reviewer: Names of things - is it maturin, `maturin`, or Maturin? And similarly pyo3, `pyo3`, or PyO3?

I've often wondered about this, I would be tempted to update pyo3's documentation to match...

Reviewer 2: I have kinda given up on using the capitalization of the various projects and just write everything lowercase except at the beginning of a sentence, so ¯\\\_(ツ)\_/¯

For pyo3 I wouldn't mind just dropping at least the capital O; I know it's a chemisty pun but in code you only use the lowercase version anyway.

detected. You can also pass `-b` / `--bindings` command line option to manually
specify it.

## `pyo3`

[pyo3](https://github.com/PyO3/pyo3) is Rust bindings for Python,
including tools for creating native Python extension modules.
It supports both CPython and PyPy.

> maturin automatically detects pyo3 bindings when it's added as a dependency in `Cargo.toml`.

Reviewer: Could be more specific here

```suggestion
maturin automatically detects pyo3 bindings when pyo3 is added as a dependency in `Cargo.toml`.
```

### `Py_LIMITED_API`/abi3

> pyo3 bindings has `Py_LIMITED_API`/abi3 support, enable the `abi3` feature of the `pyo3` crate to use it:

Reviewer: ```suggestion
pyo3 bindings can be built for `Py_LIMITED_API`/abi3. Enable the `abi3` feature of the `pyo3` crate to use it:
```

```toml
pyo3 = { version = "0.14", features = ["abi3"] }
```

> > **Note**: Read more about abi3 support in [pyo3's documentation](https://pyo3.rs/latest/building_and_distribution.html#py_limited_apiabi3).

Reviewer: I think this sentence might want to be placed _before_ telling users how to enable `abi3`, so that they understand what it is.
